packages/starknet-snap/src/ui/utils.test.tsx

```diff
@@ -0,0 +1,101 @@
+import type { FormattedCallData } from '../types/snapState';
+import { DEFAULT_DECIMAL_PLACES } from '../utils/constants';
+import { accumulateTotals } from './utils';
+
+describe('calculateTotals', () => {
+  const mockCalls = (overrides = [{}]) =>
+    [
+      {
+        tokenTransferData: {
+          amount: '1000000000000000000', // 1 ETH as string BigInt
+          symbol: 'ETH',
+          decimals: 18,
+          ...overrides[0],
+        },
+      },
+      {
+        tokenTransferData: {
+          amount: '500000000000000000', // 0.5 ETH as string BigInt
+          symbol: 'ETH',
+          decimals: 18,
+          ...overrides[1],
+        },
+      },
+      {
+        tokenTransferData: {
+          amount: '2000000000000000000', // 2 STRK as string BigInt
+          symbol: 'STRK',
+          decimals: 18,
+          ...overrides[2],
+        },
+      },
+    ] as FormattedCallData[];
+
+  const mockMaxFee = '100000000000000000'; // 0.1 token fee
+
+  it('sums up transfer amounts for each token', () => {
+    const calls = mockCalls();
+    const selectedFeeToken = 'ETH';
+
+    const result = accumulateTotals(calls, mockMaxFee, selectedFeeToken);
+
+    expect(result).toStrictEqual({
+      ETH: {
+        amount: BigInt('1600000000000000000'), // 1 + 0.5 + 0.1 ETH
+        decimals: 18,
+      },
+      STRK: {
+        amount: BigInt('2000000000000000000'), // 2 STRK
+        decimals: 18,
+      },
+    });
+  });
+
+  it('handles fee-only transactions when there are no transfers', () => {
+    const calls = mockCalls();
+    calls[0].tokenTransferData = undefined;
+    calls[1].tokenTransferData = undefined;
+    calls[2].tokenTransferData = undefined;
+    const selectedFeeToken = 'STRK';
+    const result = accumulateTotals(calls, mockMaxFee, selectedFeeToken);
+
+    expect(result).toStrictEqual({
+      STRK: {
+        amount: BigInt('100000000000000000'), // 0.1 STRK
+        decimals: DEFAULT_DECIMAL_PLACES,
+      },
+    });
+  });
+
+  it('creates a new token entry if the fee token was not part of calls', () => {
```

Comment:
this test is not valid, as your calls already include STRK

and i think you can combine with `sums up transfer amounts for each token` with it.each

Reply:
we should remove this test afterward